### PR TITLE
Feature/2649 2nd approach

### LINKED ISSUE #2649: Refactor rootStore.state.config to dedicated module
## What is the motivation for adding / enhancing this feature?

The `rootStore.state.config.entities.attribute.includeFields` kind of paths are all around the code. We should create a dedicated module `ConfigManager` (like [RouteManager](https://github.com/DivanteLtd/vue-storefront/blob/develop/core/lib/router-manager.ts)) to keep the config.

Note: The thing is that the config [needs to be updated runtime](https://github.com/DivanteLtd/vue-storefront/blob/1b51682feb55cc9098d3d36a5a0e47505b827472/core/app.ts#L72)

## What are the acceptance criteria 

- the old pattern `rootStore.state.config` should still work for backward compatibility
- all the instances in the core + theme of `rootStore.state.config` replaced by this new fancy module

### PR DESCRIPTION
**Pull request by pkarw**
### Related issues

closes #2649 

### Short description and why it's useful

I've refactored the way we access the config. The `rootStore.state.config` is still available however - deprecated. I believe we should remove this property starting with Vue Storefront 2.0.

From 1.10.rc-1 the only recommended way to access the config file is just:
```js
import config from `config`
```

I've tested the code with the `config.dynamicConfigReload=true` which provides the user with the modified config files within each request.

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: config/default.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
       "outputCacheDefaultTtl": 86400,
       "availableCacheTags": ["product", "category", "home", "checkout", "page-not-found", "compare", "my-account", "P", "C", "error"],
       "invalidateCacheKey": "aeSu7aip",
-      "dynamicConfigReload": false,
```
Comment (filrak) [inline config/default.json]: shouldn't be "false" by default?

File: core/compatibility/plugins/config/index.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
-import store from '@vue-storefront/core/store'
-const config = store.state.config
+import config from 'config'
```
Comment (filrak) [inline core/compatibility/plugins/config/index.js]: in some places you use `globalConfig` and in some `config` i think we should just have named export instead of default one so the naming is consistant everywhere. It's generally recomended to use named exports

File: core/modules/cart/store/mutations.ts [hunk 3]
```diff
@@ -67,8 +67,8 @@ const mutations: MutationTree<CartState> = {
     state.cartIsLoaded = true
     state.cartSavedAt = Date.now()
 
-    // Vue.prototype.$bus.$emit('order/PROCESS_QUEUE', { config: rootStore.state.config }) // process checkout queue
-    Vue.prototype.$bus.$emit('sync/PROCESS_QUEUE', { config: rootStore.state.config }) // process checkout queue
+    // Vue.prototype.$bus.$emit('order/PROCESS_QUEUE', { config: config }) // process checkout queue
+    Vue.prototype.$bus.$emit('sync/PROCESS_QUEUE', { config: config }) // process checkout queue
```
Comment (filrak) [inline core/modules/cart/store/mutations.ts]:
```suggestion
    Vue.prototype.$bus.$emit('sync/PROCESS_QUEUE', { config }) // process checkout queue
```

File: core/modules/cart/test/unit/store/actions.spec.ts [hunk 4]
```diff
@@ -80,9 +80,10 @@ describe('Cart actions', () => {
     wrapper(cartActions);
```
Comment (filrak) [inline core/modules/cart/test/unit/store/actions.spec.ts]: why you made all of those sync?

File: core/helpers/index.ts [hunk 5]
```diff
@@ -1,10 +1,8 @@
-import rootStore from '@vue-storefront/core/store'
 import SearchQuery from '@vue-storefront/core/lib/search/searchQuery'
 import { remove as removeAccents } from 'remove-accents'
-import { Logger } from '@vue-storefront/core/lib/logger'
 import { formatCategoryLink } from '@vue-storefront/core/modules/url/helpers'
 import Vue from 'vue'
-import config from 'config'
+import config from 'config';
```
Comment (patzick) [inline core/helpers/index.ts]:
```suggestion
import config from 'config'
```
i believe we would have linter problem here

File: core/compatibility/plugins/config/index.js (hunk 2 above)
Comment (pkarw) [reply core/compatibility/plugins/config/index.js]: I don't think it provides us any value and requires a lot of changes in the docs etc. Can't apply this one.

File: core/modules/cart/test/unit/store/actions.spec.ts (hunk 4 above)
Comment (pkarw) [reply core/modules/cart/test/unit/store/actions.spec.ts]: Haven't modified the tests

File: config/default.json (hunk 1 above)
Comment (pkarw) [reply config/default.json]: Yes, it shoulld. thanks

**Comment by pkarw** [comment]
The only thing in which https://github.com/DivanteLtd/vue-storefront/pull/2889 is superior over this PR is that currently to alter the config (for example to test some edge cases in unit tests) You can do sth like:

```js
import config from `config`
Object.assign(config, { yourLocalModifications: `just for unit tests` })
// then run the test logic
```

in case of the `ConfigManager` that has been introduced in the #2889 You might have done it just by:

```js
ConfigManager.updateConfig({ yourLocalModifications: `just for unit tests` })
```

.. it was "more like IoC" however this is the end of pros list :) 

So I like this PR more. We can do kind of injecting the `updateConfig` function into `config` module runtime to be able to alter the config more elegant way - not sure if it's required for now.

I'm explaining this just for a record
